### PR TITLE
MULE-17121: Upgrade maven-site-plugin to 3.8.2

### PR DESCRIPTION
**Pull request by lvzh90**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -145,8 +145,9 @@
         <maven.compiler.plugin.version>3.8.0</maven.compiler.plugin.version>
         <maven.install.plugin.version>2.5.2</maven.install.plugin.version>
         <maven.deploy.plugin.version>2.8.2</maven.deploy.plugin.version>
-        <maven.site.plugin.version>3.7.1</maven.site.plugin.version>
-        <maven.javadoc.plugin.version>3.0.1</maven.javadoc.plugin.version>
+        <xDocLint>-Xdoclint:none</xDocLint>
+        <maven.site.plugin.version>3.8.2</maven.site.plugin.version>
+        <maven.javadoc.plugin.version>3.1.0</maven.javadoc.plugin.version>
```
Comment (aiannucci) [inline pom.xml]: Use 3.1.1

File: pom.xml [hunk 2]
```diff
@@ -566,6 +567,7 @@
                             </execution>
                         </executions>
                         <configuration>
+                            <doclint>none</doclint>
```
Comment (aiannucci) [inline pom.xml]: What is the difference between this `doclint` parameter and the one used in the `additionalparam` parameter, in the line below this one?

Comment (lvzh90) [reply pom.xml]: If you are running from maven with maven-javadoc-plugin v3.0.0 or later, you need to use the doclint setting.
When using earlier versions of maven-javadoc-plugin (before v3.0.0), you need to use a different setting: <additionalparam>-Xdoclint:none</additionalparam>

File: pom.xml (hunk 1 above)
Comment (lvzh90) [reply pom.xml]: Done

**Pull request merged**